src/Chapter3.hs

```diff
@@ -374,6 +388,25 @@ after the fight. The battle has the following possible outcomes:
 
 -}
 
+data Character = MkCharacter
+  { characterHealth   :: Int
+  , characterAttack   :: Int
+  , characterGold     :: Int
+  } deriving (Show)
+
+fight :: Character -> Character -> Character
```

Comment:
The only issue with the current implementation and reusages of the same data type for `Monster` and `Knight` in here is that you can easily mix them when calling the function:

```haskell
> fight knight monster
> fight monster knight
```

Both of the calls are absolutely valid, but the results are going to be different! Because your implementation expects a particular order of arguments, but it's so easy to provide a wrong order.

Here, we had in mind to make two type for each Monster and Knight. But as you already noticed, these two would be almost identical, and it feels wrong to duplicate that. Later we show the `newtypes` which usually resolve these issues (we didn't expect you to use them in here, this is just the additional note 🙂 )
```haskell
newtype Monster = Monster Entity
newtype Knight = Knight Entity
```